### PR TITLE
Dokumentation started

### PR DESCRIPTION
**Pull request by hendrikebbers**
Added documentation:
- general build documentation
- general project structure
- documenation best practices

Oh, it looks like that this PR contains an additional fix that @blackdrag @sclassen and I did but never committed :D

### CODE REVIEW

File: remoting/rico-remoting-common/src/test/java/dev/rico/internal/remoting/converters/ZonedDateTimeConverterTest.java [hunk 1]
```diff
@@ -1,127 +1,152 @@
...
-            final ZonedDateTime reconvertedTime = (ZonedDateTime) reconverted;
-            Assert.assertEquals(ZoneId.of("Australia/Sydney"), reconvertedTime.getZone());
-
-        } finally {
-            TimeZone.setDefault(defaultZone);
-        }
-    }
-
-}
+/*
```
Comment (sclassen) [inline remoting/rico-remoting-common/src/test/java/dev/rico/internal/remoting/converters/ZonedDateTimeConverterTest.java]: what did change here ??

Comment (madmas) [reply remoting/rico-remoting-common/src/test/java/dev/rico/internal/remoting/converters/ZonedDateTimeConverterTest.java]: I think the change to the java file should be moved to a separate PR

Comment (hendrikebbers) [reply remoting/rico-remoting-common/src/test/java/dev/rico/internal/remoting/converters/ZonedDateTimeConverterTest.java]: will do

File: documentation/src/docs/asciidoc/build-doc/project-structure.adoc [hunk 2]
```diff
@@ -0,0 +1,69 @@
+ifndef::imagesdir[:imagesdir: ../images]
+
+== Project structure
+
+Rico is structured in several modules. All these modules are defined as gradle subprojects, 
+which will create a jar per module. Next to this each module is defined
```
Comment (hendrikebbers) [inline documentation/src/docs/asciidoc/build-doc/project-structure.adoc:6]:
```suggestion
which will create a jar per huhu. Next to this each module is defined
```

File: documentation/src/docs/asciidoc/build-doc/project-structure.adoc [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+(see folder integration-tests). All these modules may not follow all the rules and
+structure defined here.
+====
+
+=== Gradle build
+
+In the root folder of the project you can find the `build.gradle` file that defines
+the build for the rico project and all subprojects. Instead of externalizing build
+tasks in the subprojects we decided to define the build steps for different subprojects
+types (java module or documentation) in this central file.
```
Comment (hendrikebbers) [inline documentation/src/docs/asciidoc/build-doc/project-structure.adoc:22]:
```suggestion
types (java module or documentation) in this central file.
njnuj
```

**Pull request merged**